### PR TITLE
Zims further gulp common updates

### PR DESCRIPTION
**Pull request by zikalino**
- added license in files
- raspberry pi c major updates
- fixed problem with error handling in scp file transfer
- preparing ssh function to handle errors by checking for marker in output stream

### CODE REVIEW

**Comment by zikalino** [comment]
@aovechki @ZhijunZhao

File: all.js [hunk 1]
```diff
@@ -13,118 +15,34 @@ var biHelper = require('./biHelper.js');
...
- * @param {string[]} targetFileList - List of files at destination
- * @param {callback} callback - Callback
- */
-function uploadFilesViaScp(config, sourceFileList, targetFileList, callback)
+function uploadFilesViaScp(config, sourceFileList, targetFileList, cb)
 {
-  if(sourceFileList.length == 0) return;
-  
+  if(sourceFileList.length == 0) {
+    cb();
```
Comment (ZhijunZhao) [inline all.js]: if(cb) cb();

File: all.js [hunk 2]
```diff
@@ -13,118 +15,34 @@ var biHelper = require('./biHelper.js');
  * @param {} config
  * @param {string[]} sourceFileList - List of local files
  * @param {string[]} targetFileList - List of files at destination
- * @param {callback} callback - Callback
+ * @param {cb} callback - Callback
```
Comment (ZhijunZhao) [inline all.js]: @param {callback}  cb

File: all.js [hunk 3]
```diff
@@ -211,27 +129,53 @@ function localClone(url, folder, verbose, cb) {
...
+      if (options && options.verbose) {
+        process.stdout.write(o);
+      }
+
+      output += String(o);
+    },
+    exit: function() {
+      // setting short timeout, as exit handler may be called before remaining data
+      // arrives via out
+        console.log("BEFORE TIMEOUT");
```
Comment (ZhijunZhao) [inline all.js]: As discussed, please remove it.

File: all.js [hunk 4]
```diff
@@ -211,27 +129,53 @@ function localClone(url, folder, verbose, cb) {
...
+          if (output.indexOf(options.marker) < 0) {
+            var err = new Error("SSH command hasn't completed successfully");
+            err.stack = err.message;
+            err.marker = true;
+            cb(err);
+            return;
+          }
+        }
+        console.log("AFTER TIMEOUT");
+        cb();
```
Comment (ZhijunZhao) [inline all.js]: If (cb) check?

File: all.js [hunk 5]
```diff
@@ -286,18 +230,18 @@ function folderExistsSync(path) {
  * @param {callback}  successCB
  * @param {callback}  failureCB
  */
```
Comment (ZhijunZhao) [inline all.js:229]: update the param notes

File: all.js [hunk 6]
```diff
@@ -309,24 +253,24 @@ function download(srcZipUrl, targetZipPath, successCB, failureCB)
...
```
Comment (ZhijunZhao) [inline all.js:251]: update param notes.

File: all.js (hunk 1 above)
Comment (zikalino) [reply all.js]: ok fixed everywhere

File: all.js (hunk 2 above)
Comment (zikalino) [reply all.js]: fixed

File: all.js (hunk 3 above)
Comment (zikalino) [reply all.js]: fixed

File: all.js (hunk 4 above)
Comment (zikalino) [reply all.js]: fixed

File: all.js (hunk 6 above)
Comment (zikalino) [reply all.js:251]: fixed

**Review by ZhijunZhao — APPROVED** [review]

**Pull request merged**